---
title: "Major Histocompatibility Complex"
author: "Nicolas Brosse"
date: "2024-mm-dd"
categories: [biology]
bibliography: mhc-references.bib
description: "MHC"
image: figures/cover.jpg
---

# Major histocompatibility complex

::: callout-note
## This section (text and figures) is drawn from the excellent wikipedia page https://en.wikipedia.org/wiki/Major_histocompatibility_complex
:::

The major histocompatibility complex (MHC) is a large region on vertebrate DNA that contains a set of closely linked, highly variable genes responsible for coding cell surface proteins essential for the adaptive immune system. These cell surface proteins are known as MHC molecules.

The primary function of MHC molecules is to bind antigens derived from self-proteins or pathogens and present them on the cell surface for recognition by the appropriate T-cells. MHC molecules facilitate interactions between leukocytes (white blood cells) and other cells, playing a critical role in the immune response. They are also crucial in determining donor compatibility for organ transplants and influence susceptibility to autoimmune diseases.

Within a cell, protein molecules from the host or other biological entities are continuously synthesized and degraded. Each MHC molecule on the cell surface presents a small peptide fragment, called an **epitope**, from these proteins. The presentation of self-antigens helps prevent the immune system from attacking the body's own cells, while pathogen-derived antigen presentation leads to the destruction of infected cells by the immune system.

The diversity of self-antigen presentation by MHC molecules is achieved in three main ways: (1) the MHC repertoire is polygenic, involving multiple interacting genes; (2) MHC expression is codominant, meaning alleles from both parents are expressed; and (3) MHC genes are highly polymorphic, with significant variation among individuals within a species. Additionally, MHC I presentation may involve antigenic peptide splicing, combining peptides from different proteins to further increase antigen diversity.

The MHC locus exists in all jawed vertebrates. Despite variations in the number of genes among different species, the overall organization of the locus is similar. Typically, the MHC region includes about a hundred genes and pseudogenes, though not all are involved in immunity. In humans, the MHC region is located on chromosome 6 and contains 224 genes spanning 3.6 megabase pairs (3,600,000 bases), with about half having known immune functions. The human MHC is also known as the HLA (human leukocyte antigen) complex. Other species have similar designations, such as SLA (swine leukocyte antigens) in pigs, BoLA in cattle, and DLA in dogs.

The MHC gene family is divided into three subgroups: MHC class I, MHC class II, and MHC class III. Among these, the genes coding for MHC class I and class II molecules are directly involved in antigen presentation and are highly polymorphic. For humans, the IMGT database lists 27301 alleles for class I HLA and 11674 for class II HLA (in 2024).

## MHC class I

MHC class I molecules are expressed on nearly all nucleated cells, including platelets, but not on red blood cells. They present epitopes (8–10 amino acids residues) to cytotoxic T lymphocytes (CTLs), which express CD8 receptors and T-cell receptors (TCRs). When a CTL’s CD8 receptor binds to an MHC class I molecule, and if the CTL's TCR recognizes the epitope presented by the MHC class I molecule, the CTL induces programmed cell death (apoptosis) in the target cell, see @fig-schematic-view-MHC-T-cell and @fig-T-cell-MHC. Thus, MHC class I molecules are crucial for mediating cellular immunity, primarily targeting intracellular pathogens such as viruses and some bacteria. In humans, MHC class I molecules include HLA-A, HLA-B, and HLA-C.

[@antunes2018]

![Schematic view of the role of MHCs in T-cell activation. Class I Major Histocompatibility Complexes (MHC-I) are present in almost every cell and involved in the surface presentation of peptides derived from intracellular proteins. On the other hand, class II MHCs are present only in "professional" antigen-presenting cells (phagocytes) and involved in the surface presentation of peptides derived from extracellular proteins. The recognition of displayed peptideMHC complexes by the T-cell receptor (TCR) triggers T-cell activation, clonal expansion and immunological memory. While cytotoxic T-cells (CD8+) mediate cellular immunity, helper T-cells (CD4+) control the humoral response and have other regulatory roles. CD stands for cluster of differentiation. [@antunes2018]](figures/Schematic-view-of-the-role-of-MHCs-in-T-cell-activation-Class-I-Major-Histocompatibility_W640.jpg){#fig-schematic-view-MHC-T-cell}

![T-cell receptor complexed with MHC-I and MHC-II.](figures/T-CellReceptor-MHC.jpg){#fig-T-cell-MHC}

The crystal structure of an MHC class I molecule reveals that MHC class I molecules are heterodimers, consisting of a polymorphic heavy α-subunit encoded within the MHC locus and an invariant β2 microglobulin subunit typically encoded outside the MHC locus, see @fig-HLA-class-I-schematic-structure and @fig-HLA-class-I-structure. The polymorphic heavy chain features an N-terminal extracellular region with three domains (α1, α2, and α3), a transmembrane helix anchoring the molecule to the cell surface, and a short cytoplasmic tail. The α1 and α2 domains form a deep peptide-binding groove with two long α-helices and a floor composed of eight β-strands. The α3 domain interacts with the CD8 co-receptor. The β2 microglobulin subunit stabilizes the complex and aids in the recognition of the peptide-MHC class I complex by the CD8 co-receptor. The peptide is non-covalently bound within the MHC class I molecule's groove, held in place by several pockets. The most polymorphic amino acid side-chains in human alleles occupy the central, wider part of the binding groove, while conserved side-chains cluster at its narrower ends.

[@feng2014]

![HLA class I schematic structure, source [@feng2014].](figures/MHC_class_I_structure.webp){#fig-HLA-class-I-schematic-structure}

![HLA class I structure, source [@feng2014].](figures/HLA-class-I-structure.webp){#fig-HLA-class-I-structure}

## MHC class II

MHC class II molecules can be conditionally expressed by all cell types but are typically found on "professional" antigen-presenting cells (APCs), such as macrophages, B cells, and especially dendritic cells (DCs). These APCs uptake antigenic proteins, process them, and display a fragment of the protein, known as an epitope (13–18 amino acids residues), on their surface bound to an MHC class II molecule—a process known as antigen presentation. On the cell surface, the epitope can be recognized by T-cell receptors (TCRs), with the specific region that binds to the epitope called the paratope, see @fig-schematic-view-MHC-T-cell and @fig-T-cell-MHC.

Helper T cells have CD4 receptors and TCRs on their surface. When a naive helper T cell's CD4 receptor binds to an APC's MHC class II molecule, its TCR can then bind to the epitope presented by the MHC class II. This interaction primes the naive T cell. Depending on the local environment and the balance of cytokines secreted by APCs, the naive helper T cell (Th0) can differentiate into either a memory Th cell or an effector Th cell of one of several phenotypes: type 1 (Th1), type 2 (Th2), type 17 (Th17), or regulatory/suppressor (Treg) cells.

Thus, MHC class II molecules mediate the immune response to an antigen or, if APCs primarily induce differentiation into Treg cells, immune tolerance of the antigen. The polarization of Th0 cells during the initial exposure to an antigen is crucial in determining the course of chronic diseases, such as inflammatory bowel disease and asthma, by influencing the immune response coordinated by memory Th cells upon subsequent exposure to similar antigens. B cells also express MHC class II molecules to present antigens to Th0 cells. When B cell receptors bind matching epitopes, they activate and secrete antibodies, mediating humoral immunity.

MHC class II molecules are heterodimers composed of polymorphic α and β subunits, with genes located within the MHC class II subregion. The peptide-binding groove of MHC class II molecules is formed by the N-terminal domains of both the α1 and β1 subunits, unlike MHC class I molecules, which involve two domains from the same chain. Both subunits of MHC class II molecules also contain a transmembrane helix and immunoglobulin domains (α2 and β2), which can be recognized by CD4 co-receptors. This specificity ensures that different lymphocytes bind to the appropriate antigens with high affinity, see @fig-MHC-II-crystal-structure.

[@feng2014]

![Crystal structure of MHC II determined by X-ray crystallography [@feng2014]](figures/Color-online-Crystal-structure-of-MHC-II-determined-by-X-ray-crystallography-15-a-A_W640.jpg){#fig-MHC-II-crystal-structure}

[@antunes2018]

![Molecular structures of class I and class II MHCs. Molecular representation of a class I MHC (A, C) and a class II MHC (B, D). The upper panel shows a top view, while the bottom panel shows a cross section side-view of the binding clefts. Note that the binding cleft of a class I receptor is deeper, with "closed" extremities, while the class II cleft is shallower, with open extremities. The pockets involved in binding primary "anchor" residues are indicated. Together, structural differences in the shape of the cleft and the location of binding pockets have an impact on the overall conformation of bound ligands (e.g., peptides tend to adopt bulged conformations when bound to class I, and more linear conformations when bound to class II). Crystal structures of both complexes were downloaded from the PDB and superimposed to be in the same orientation. Class I complex: HLA-A\*01:01 receptor presenting a tumor-derived 9-mer peptide (PDB code 5BRZ). Class II complex: HLA-DRB1\*01:01 receptor presenting a 14-mer bacteria-derived peptide (PDB code 1KLU). [@antunes2018]](figures/Molecular-structures-of-class-I-and-class-II-MHCs-Molecular-representation-of-a-class-I_W640.jpg){#fig-mol-structures-I-II-MHC}

+---------------------------------+-----------------------------------------+------------------------------------------+
| Feature                         | Class I MHC                             | Class II MHC                             |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Constituting polypeptide chains | α chain (45KDa in humans)               | α chain (30–34 KDa in humans)            |
|                                 |                                         |                                          |
|                                 | β~2~ chain (12 KDa in humans)           | β chain (26–29 KDa in humans)            |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Antigen binding domain          | α1and α2 domains                        | α~1~ and β~1~ domains                    |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Binds protein antigens of       | 8–10 amino acids residues               | 13–18 amino acids residues               |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Peptide bending cleft           | Floor formed by β sheets and sides by α | Floor formed by β sheets and sides by α  |
|                                 |                                         |                                          |
|                                 | helices, blocked at both the ends       | helices, opened at both the ends         |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Antigenic peptide motifs        | Anchor residues located at amino and    | Anchor residues located almost uniformly |
|                                 |                                         |                                          |
| involved in binding             | carbon terminal ends                    | along the peptide                        |
+---------------------------------+-----------------------------------------+------------------------------------------+
| Presents antigenic peptide to   | CD8+ T cells                            | CD4+ T cells                             |
+---------------------------------+-----------------------------------------+------------------------------------------+

![MHC class I pathway: Proteins in the cytosol are degraded by the proteasome, liberating peptides internalized by TAP channel in the endoplasmic reticulum, there associating with MHC-I molecules freshly synthesized. MHC-I/peptide complexes enter Golgi apparatus, are glycosylated, enter secretory vesicles, fuse with the cell membrane, and externalize on the cell membrane interacting with T lymphocytes.](figures/MHC_Class_I_processing.png){#fig-mhc-class-I-pathway}

![Peptide binding for Class I and Class II MHC molecules, showing the binding of peptides between the alpha-helix walls, upon a beta-sheet base. The difference in binding positions is shown. Class I primarily makes contact with backbone residues at the Carboxy and amino terminal regions, while Class II primarily makes contacts along the length of the residue backbone. The precise location of binding residues is determined by the MHC allele.](figures/MHC_Binding_Diagram.png)

## Human leukocyte antigen

[Detailed wikipedia page](https://en.wikipedia.org/wiki/Human_leukocyte_antigen)

![](figures/MHC_expression.png)

Human MHC class I and II are also called [human leukocyte antigen](https://en.wikipedia.org/wiki/Human_leukocyte_antigen "Human leukocyte antigen") (HLA). To clarify the usage, some of the biomedical literature uses HLA to refer specifically to the HLA protein molecules and reserves MHC for the region of the genome that encodes for this molecule, but this is not a consistent convention.

The most studied HLA genes are the nine classical MHC genes: [*HLA-A*](https://en.wikipedia.org/wiki/HLA-A "HLA-A")*, [HLA-B](https://en.wikipedia.org/wiki/HLA-B "HLA-B"), [HLA-C](https://en.wikipedia.org/wiki/HLA-C "HLA-C"), [HLA-DPA1](https://en.wikipedia.org/wiki/HLA-DPA1 "HLA-DPA1"), [HLA-DPB1](https://en.wikipedia.org/wiki/HLA-DPB1 "HLA-DPB1"), [HLA-DQA1](https://en.wikipedia.org/wiki/HLA-DQA1 "HLA-DQA1"), [HLA-DQB1](https://en.wikipedia.org/wiki/HLA-DQB1 "HLA-DQB1"), [HLA-DRA](https://en.wikipedia.org/wiki/HLA-DRA "HLA-DRA")*, and [*HLA-DRB1*](https://en.wikipedia.org/wiki/HLA-DRB1 "HLA-DRB1"). In humans, the MHC gene cluster is divided into three regions: classes I, II, and III. The A, B and C genes belong to MHC class I, whereas the six D genes belong to class II.

MHC alleles are expressed in codominant fashion.^[\[60\]](https://en.wikipedia.org/wiki/Major_histocompatibility_complex#cite_note-ABBAS-60)^ This means the [alleles](https://en.wikipedia.org/wiki/Allele "Allele") (variants) inherited from both parents are expressed equally:

-   Each person carries 2 alleles of each of the 3 class-I genes, (*HLA-A, HLA-B* and *HLA-C*), and so can express six different types of MHC-I (see figure).

-   

-   In the class-II locus, each person inherits a pair of HLA-DP genes (DPA1 and DPB1, which encode α and β chains), a couple of genes *HLA-DQ* (*DQA1* and *DQB1*, for α and β chains), one gene *HLA-DRα* (*DRA1*), and one or more genes *HLA-DRβ* (*DRB1* and *DRB3, -4* or *-5*). That means that one [heterozygous](https://en.wikipedia.org/wiki/Heterozygous "Heterozygous") individual can inherit six or eight functioning class-II alleles, three or more from each parent. The role of *DQA2* or *DQB2* is not verified. The *DRB2, DRB6, DRB7, DRB8* and *DRB9* are pseudogenes.

The set of alleles that is present in each chromosome is called the MHC [haplotype](https://en.wikipedia.org/wiki/Haplotype "Haplotype"). In humans, each HLA allele is named with a number. For instance, for a given individual, his haplotype might be HLA-A2, HLA-B5, HLA-DR3, etc... Each heterozygous individual will have two MHC haplotypes, one each from the paternal and maternal chromosomes.

The MHC genes are highly polymorphic; many different alleles exist in the different individuals inside a population. The polymorphism is so high, in a mixed population (non[endogamic](https://en.wikipedia.org/wiki/Endogamy "Endogamy")), no two individuals have exactly the same set of MHC molecules, with the exception of [identical twins](https://en.wikipedia.org/wiki/Twin "Twin").

The polymorphic regions in each allele are located in the region for peptide contact. Of all the peptides that could be displayed by MHC, only a subset will bind strongly enough to any given HLA allele, so by carrying two alleles for each gene, each encoding specificity for unique antigens, a much larger set of peptides can be presented.

On the other hand, inside a population, the presence of many different alleles ensures there will always be an individual with a specific MHC molecule able to load the correct peptide to recognize a specific microbe. The evolution of the MHC polymorphism ensures that a population will not succumb to a new pathogen or a mutated one, because at least some individuals will be able to develop an adequate immune response to win over the pathogen. The variations in the MHC molecules (responsible for the polymorphism) are the result of the inheritance of different MHC molecules, and they are not induced by [recombination](https://en.wikipedia.org/wiki/Genetic_recombination "Genetic recombination"), as it is the case for the antigen [receptors](https://en.wikipedia.org/wiki/Immune_receptor "Immune receptor").

# Methods

<https://services.healthtech.dtu.dk/services/NetMHCpan-4.1/>

IPD-IMGT Database

<https://www.ebi.ac.uk/ipd/imgt/hla/download/>

<https://www.ebi.ac.uk/ipd/imgt/hla/alleles/allele/?accession=HLA00006>

iedb database

<https://www.iedb.org/home_v3.php>

Methods

MHCflurry

<https://github.com/openvax/mhcflurry>

<https://www.sciencedirect.com/science/article/abs/pii/S1046202324001324>

<https://github.com/GfellerLab/MixMHCpred>

<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11092674/pdf/nihpp-2024.05.01.592062v1.pdf>

<https://pubmed.ncbi.nlm.nih.gov/31578220/>

Translation

<https://www.nature.com/scitable/topicpage/translation-dna-to-mrna-to-protein-393/>

<https://www.khanacademy.org/science/ap-biology/gene-expression-and-regulation/translation/a/translation-overview>

<https://www.nature.com/scitable/topicpage/ribosomes-transcription-and-translation-14120660/>